framework/PostProcessors/DataMining.py

```diff
@@ -481,64 +485,41 @@ def collectOutput(self, finishedJob, outputObject):
-    ## If the input data object is different from the output data object, we
-    ## need to explicitly copy everything over that the user requests from the
-    ## input object
-    if inputObject != outputObject:
-      if inputObject.type != outputObject.type:
-        self.raiseAnError(IOError, 'The type of output data object ', outputObject.name, ' is not consistent with the type of input data object ', self.inputObject, '! Please check your input file')
-      rlzs = {}
-      inputData = inputObject.asDataset()
-      ## First copy any data you need from the input object
-      ## before adding new data
-      for key in outputObject.getVars('Input')+outputObject.getVars('Output'):
-        if key in inputObject.getVars('Input') + inputObject.getVars('Output'):
-          rlzs[key] = copy.deepcopy(inputObject.getVarValues(key).values)
-        else:
-          missingKeys.append(key)
-      if outputObject.type == 'PointSet':
-        outputObject.load(rlzs, style='dict')
-      elif outputObject.type == 'HistorySet':
-        rlzs[self.pivotParameter] = np.atleast_1d(self.pivotVariable)
-        outputObject.load(rlzs, style='dict', dims=inputObject.getDimensions())
-      else:
-        self.raiseAnError(IOError, 'Unrecognized type for output data object ', outputObject.name, '! Available type are HistorySet or PointSet!')
-    ## End data copy
-    ############################################################################
-
-    ############################################################################
-    ## Now augment the DataObject by adding the computed labels and/or
-    ## transformed coordinates aka the new columns added to the DataObject
-    for key in dataMineDict['outputs']:
-      if key in missingKeys:
-        missingKeys.remove(key)
-      for param in outputObject.getVars('output'):
-        if key == param:
-          outputObject.remove(variable=key)
-      if outputObject.type == 'PointSet':
-        outputObject.addVariable(key, copy.copy(dataMineDict['outputs'][key]),classify='output')
-      elif outputObject.type == 'HistorySet':
-        expValues = np.zeros(len(outputObject), dtype=object)
-        values = dataMineDict['outputs'][key]
-        for index, value in enumerate(values):
-          timeLength = len(self.pivotVariable[index])
-          arrayBase = value * np.ones(timeLength)
-          xrArray = xr.DataArray(arrayBase,dims=(self.pivotParameter), coords=[self.pivotVariable[index]])
-          expValues[index] = xrArray
-        outputObject.addVariable(key, expValues,classify='output')
-    ## End data augmentation
     ############################################################################
+    if inputObject.type != outputObject.type:
+      self.raiseAnError(IOError,"The type of output DataObject",outputObject.name,"is not consistent with input",\
+              "DataObject type, i.e. ",outputObject.type,"!=",inputObject.type)
```

Comment:
Why does this matter?

Reply:
Based on my understanding, all the data from the input data object will be copied to the output data object. That's why there is a requirement for this. This requirement is in the devel, I do not force it.

Reply:
Okay. This might need to be rethought in the future, but it's fine here.